sns/json_schemas/operation.json

```diff
@@ -0,0 +1,203 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "http://asyncapi.com/bindings/sns/operation.json",
+  "title": "Operation Schema",
+  "description": "This object contains information about the operation representation in SNS.",
+  "type": "object",
+  "additionalProperties": false,
+  "patternProperties": {
+    "^x-[\\w\\d\\.\\-\\_]+$": {
+      "$ref": "https://raw.githubusercontent.com/asyncapi/spec-json-schemas/v2.14.0/schemas/2.4.0.json#/definitions/specificationExtension"
+    }
+  },
+  "properties": {
+    "topic": {
+      "$ref": "#/definitions/identifier",
+      "description": "Often we can assume that the SNS Topic is the channel name-we provide this field in case the you need to supply the ARN, or the Topic name is not the channel name in the AsyncAPI document."
+    },
+    "consumers": {
+      "type": "array",
+      "description": "The protocols that listen to this topic and their endpoints.",
+      "items": {
+        "$ref": "#/definitions/consumer"
+      },
+      "minItems": 1
+    },
+    "deliveryPolicy": {
+      "$ref": "#/definitions/deliveryPolicy",
+      "description": "Policy for retries to HTTP. The field is the default for HTTP receivers of the SNS Topic which may be overridden by a specific consumer."
+    },
+    "bindingVersion": {
+      "type": "string",
+      "description": "The version of this binding.",
+      "default": "latest"
+    }
+  },
+  "required": [
+    "consumers"
+  ],
+  "definitions": {
+    "identifier": {
+      "type": "object",
+      "properties": {
+        "url": {
+          "type": "string",
+          "description": "The endpoint is a URL."
+        },
+        "email": {
+          "type": "string",
+          "description": "The endpoint is an email adress."
+        },
+        "phone": {
+          "type": "string",
+          "description": "The endpoint is a phone number."
+        },
+        "arn": {
+          "type": "string",
+          "description": "The target is an ARN. For example, for SQS, the identifier may be an ARN, which will be of the form: arn:aws:sqs:{region}:{account-id}:{queueName}"
+        },
+        "name": {
+          "type": "string",
+          "description": "The endpoint is identified by a name, which corresponds to an identifying field called 'name' of a binding for that protocol on this publish Operation Object. For example, if the protocol is 'sqs' then the name refers to the name field sqs binding. We don't use $ref because we are referring, not including."
+        }
+      }
+    },
```

Comment:
Looking at the AWS documentation for redrive policies it doesn't seem like the target resource to be used for redriving messages is an object but is instead a string that could be an ARN, URL, Name etc.

For example in cloud formation the redrive policy definition looks like this:
```
"{ \"deadLetterTargetArn\": \"arn:aws:sqs:us-east-2:111122224444:my-dlq\", \"maxReceiveCount\": 4 }"
```

Which appears to be [fairly representative](https://awslabs.github.io/serverless-rules/rules/sqs/redrive_policy/) of how a lot of [infrastructure tooling deals with defining redrives](https://awslabs.github.io/serverless-rules/rules/sns/redrive_policy/).

Is there a reason we are diverging from the common standards for defining redrives with a separate identifier object  that has specific properties?

Reply:
In my opinion, it's better to keep the definitions as simple as possible at the beginning. If most infra tooling only uses **deadLetterTargetArn** and **maxReceiveCount** for the Redrive Policy then there is no harm in starting with that properties.

As a result, we just need a model that can be interpreted into infra and we can add less complexity by not using comprehensive definitions (if it's applicable).